extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/CodeAuthenticationMechanism.java

```diff
@@ -471,7 +470,7 @@ public void handle(AsyncResult<Void> result) {
                                                 @Override
                                                 public void accept(SecurityIdentity identity) {
                                                     // after a successful refresh, rebuild the identity and update the cookie 
-                                                    processSuccessfulAuthentication(context, configContext, token,
+                                                    processSuccessfulAuthentication(context, configContext, token, refreshToken,
```

Comment:
Lets replace `refreshToken` with `result.opaqueRefreshToken() != null ? result.opaqueRefreshToken() : refreshToken` because this check only makes sense in context of the refresh token grant processing

Reply:
LGTM